File: annealing/annealing/temp.py
"""
This module contains functions to calculate the temperature for a given step.
"""
import math


def linear_change_temp(max_temp, min_temp, steps, step):
    """
    Return the temperature for the given step.
    """
    return max_temp - (max_temp - min_temp) * (step / steps)


def logarithmic_change_temp(max_temp, min_temp, steps, step):
    """
    Return the temperature for the given step.
    """
    return max_temp / (math.log(1 + step) or 0.0001)


def change_temp(max_temp, min_temp, steps, step):
    """ "
    Return the temperature for the given step.
    """
    # return logarithmic_change_temp(max_temp, min_temp, steps, step)
    return linear_change_temp(max_temp, min_temp, steps, step)

File: src/app/groups/annealing/temp.ts
function linear_change_temp(
  max_temp: number,
  min_temp: number,
  steps: number,
  step: number,
): number {
  return max_temp - (max_temp - min_temp) * (step / steps);
}

function logarithmic_change_temp(
  max_temp: number,
  min_temp: number,
  steps: number,
  step: number,
): number {
  return max_temp / (Math.log(1 + step) || 0.0001);
}

export function change_temp(
  max_temp: number,
  min_temp: number,
  steps: number,
  step: number,
): number {
  // You can comment/uncomment the line depending on whether you want to use linear or logarithmic temperature change
  // return logarithmic_change_temp(max_temp, min_temp, steps, step);
  return linear_change_temp(max_temp, min_temp, steps, step);
}
